lib/node_modules/@stdlib/array/base/mskfilter-map/lib/assign.js

```diff
@@ -0,0 +1,245 @@
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} clbk - function to apply
+* @param {*} [thisArg] - function execution context
+* @returns {Collection} output array
+*
+* @example
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = toAccessorArray( [ 1, 2, 3, 4 ] );
+* var mask = toAccessorArray( [ 0, 1, 0, 1 ] );
+*
+* function mappingFunction( x ) {
+*     return x * 10;
+* }
+*
+* var out = toAccessorArray( [ 0, 0, 0, 0 ] );
+* var arr = accessors( arraylike2object( x ), arraylike2object( mask ), arraylike2object( out ), 1, 0, mappingFunction );
+*
+* var v = arr.get( 0 );
+* // returns 20
+*
+* v = arr.get( 1 );
+* // returns 40
+*/
+function accessors( x, mask, out, stride, offset, clbk, thisArg ) {
+	var xdata;
+	var mdata;
+	var odata;
+	var xget;
+	var mget;
+	var oset;
+	var io;
+	var i;
+
+	xdata = x.data;
+	mdata = mask.data;
+	odata = out.data;
+
+	xget = x.accessors[ 0 ];
+	mget = mask.accessors[ 0 ];
+	oset = out.accessors[ 1 ];
+
+	io = offset;
+	for ( i = 0; i < xdata.length; i++ ) {
+		if ( mget( mdata, i ) ) {
+			oset( odata, io, clbk.call( thisArg, xget( xdata, i ) ) );
+			io += stride;
+		}
+	}
+	return odata;
+}
+
+/**
+* Applies a mask to a complex array & apply mapping function and assigns unmasked values to elements in a complex output array.
+*
+* @private
+* @param {Collection} x - real-valued floating-point input array view
+* @param {Object} mask - mask array object
+* @param {Collection} out - real-valued floating-point output array view
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} clbk - function to apply
+* @param {*} [thisArg] - function execution context
+* @returns {Collection} output array view
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );
+* var mask = [ 0, 1, 0, 1 ];
+*
+* var out = new Float64Array( 8 );
+*
+* function mappingFunction( x ) {
+*     return x * 10.0;
+* }
+*
+* var arr = complex( x, arraylike2object( mask ), out, 1, 0, mappingFunction );
+* // returns <Float64Array>[ 30.0, 40.0, 70.0, 80.0, 0.0, 0.0, 0.0, 0.0 ]
+*/
+function complex( x, mask, out, stride, offset, clbk, thisArg ) {
+	var mdata;
+	var mget;
+	var io;
+	var so;
+	var i;
+	var j;
+
+	mdata = mask.data;
+	mget = mask.accessors[ 0 ];
+
+	so = stride * 2; // note: multiply by 2, as real-valued array consists of interleaved real and imaginary components
+	io = offset * 2;
+	for ( i = 0; i < mdata.length; i++ ) {
+		if ( mget( mdata, i ) ) {
+			j = i * 2;
+			out[ io ] = clbk.call( thisArg, x[ j ] );
+			out[ io+1 ] = clbk.call( thisArg, x[ j+1 ] );
```

Comment:
This isn't what we want. We don't want to invoke the callback with only a real or an imaginary component. We want to invoke with a complex number. Accordingly, I don't think we need a separate `complex(...)` function. Rather, we only need "indexed" and "accessor" implementations.

Reply:
ok, fix it soon